website/src/components/Edit/EditableSequences.ts

```diff
@@ -0,0 +1,127 @@
+import type { SequenceEntryToEdit } from '../../types/backend.ts';
+import type { ReferenceGenomesLightweightSchema } from '../../types/referencesGenomes.ts';
+
+type EditableSequenceFile = {
+    key: string;
+    label?: string;
+    value: string | null;
+    initialValue: string | null;
+};
+
+export class EditableSequences {
+    private static nextKey = 0;
+
+    private readonly editableSequenceFiles: EditableSequenceFile[];
+    private readonly maxNumberOfRows: number;
+
+    public get rows(): Required<EditableSequenceFile>[] {
+        const rows = this.editableSequenceFiles.map((row, i) => ({
+            ...row,
+            label: row.label ?? `Segment ${i + 1}`,
+        }));
+        if (rows.length < this.maxNumberOfRows) {
+            rows.push({
+                label: `+ add new sequence`,
```

Comment:
(I think we can style this thing quite a bit nicer - my mockup was intended more to illustrate the behaviour rather than a specific design -  but we don't need to block on it and I'll try to figure out something more specific)